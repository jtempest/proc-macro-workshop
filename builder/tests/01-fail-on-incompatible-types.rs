use derive_builder::Builder;

#[derive(Builder)]
pub struct EmptyCommand();

#[derive(Builder)]
pub struct TupleCommand(String, Vec<String>, Vec<String>, String);

#[derive(Builder)]
pub enum EnumCommand {
    First,
    Second(i32),
}

fn main() {}
