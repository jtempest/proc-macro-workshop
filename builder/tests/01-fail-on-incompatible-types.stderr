error: Builders may only be derived for structs with named fields.
 --> $DIR/01-fail-on-incompatible-types.rs:4:24
  |
4 | pub struct EmptyCommand();
  |                        ^^

error: Builders may only be derived for structs with named fields.
 --> $DIR/01-fail-on-incompatible-types.rs:7:24
  |
7 | pub struct TupleCommand(String, Vec<String>, Vec<String>, String);
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Builders may only be derived for structs.
  --> $DIR/01-fail-on-incompatible-types.rs:10:10
   |
10 | pub enum EnumCommand {
   |          ^^^^^^^^^^^
